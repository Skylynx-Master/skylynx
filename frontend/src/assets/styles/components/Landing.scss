@import "../Vars.scss";

.landing {
  display: grid;
  grid-area: landing;
  width: 100%;
  max-width: 750px;
  margin: 0 auto;

  grid-template-columns: 1fr;
  grid-template-rows: repeat(5, 1fr);

  grid-template-areas:
    ".   "
    "description     "
    "search   "
    "button   "
    ".   ";
}

/* specific item styles */

.description {
  grid-area: description;
  &--title {
    text-align: center;
    margin-bottom: 1rem;
  }
  &--intro {
    text-align: center;
    margin: 30px;
  }
}

.search {
  grid-area: search;

  @include center-flex;
  &--ubication {
    @include input-info;
  }
  &__info {
    text-align: center;
    padding: 40px;
  }
}
.button {
  grid-area: button;
  @include center-flex;
  &--send {
    text-align: center;
    @include button-action;
  }
}
