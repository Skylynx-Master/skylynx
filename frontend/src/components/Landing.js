import React from "react";
import "../assets/styles/components/Landing.scss";
import { Link } from "react-router-dom";

const Landing = () => {
  return (
    <section className="landing">
      <div className="description">
        <h1 className="description--title">Apartacho</h1>
        <p className="description--intro">
          {" "}
          Todos tenemos un sitio, nosotros te ayudamos a conseguirlo
        </p>
      </div>
      <div className="search">
        <p className="search__info">
          Buscar el lugar ideal es difícil y algunas veces tedioso, acá
          encontraras las mejores opciones según tus gustos, podrás contactar
          directamente al oferente ¡Sin intermediarios!,
        </p>
        {/* <input
          type="text"
          className="search--ubication"
          placeholder="Ubicacion"
        ></input> */}
      </div>
      <div className="button">
        <button type="button" className="button--send" aria-label="button">
          <Link to="/filters">Encontrar</Link>
        </button>
      </div>
      <div className="link">
        <p>
          ¿Quieres Publicar?, <Link to="/register">Regístrate</Link>
        </p>
      </div>
    </section>
  );
};

export default Landing;
