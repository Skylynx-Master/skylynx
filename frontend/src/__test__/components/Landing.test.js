import React from "react";
import Landing from "../../components/Landing";
import { mount } from "enzyme";
import { create } from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";

describe("Landing Component Testing", () => {
  const landing = mount(
    <BrowserRouter>
      <Landing />
    </BrowserRouter>
  );

  test("landing Render", () => {
    expect(landing.length).toEqual(1);
  });
});

describe("landing Snapshot", () => {
  test("Test UI Map component", () => {
    const snapmap = create(
      <BrowserRouter>
        <Landing />
      </BrowserRouter>
    );
    expect(snapmap.toJSON()).toMatchSnapshot();
  });
});
