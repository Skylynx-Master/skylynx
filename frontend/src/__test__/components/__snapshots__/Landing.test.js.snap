// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`landing Snapshot Test UI Map component 1`] = `
<section
  className="landing"
>
  <div
    className="description"
  >
    <h1
      className="description--title"
    >
      Apartacho
    </h1>
    <p
      className="description--intro"
    >
       
      Todos tenemos un sitio, nosotros te ayudamos a conseguirlo
    </p>
  </div>
  <div
    className="search"
  >
    <p
      className="search__info"
    >
      Buscar el lugar ideal es difícil y algunas veces tedioso, acá encontraras las mejores opciones según tus gustos, podrás contactar directamente al oferente ¡Sin intermediarios!,
    </p>
  </div>
  <div
    className="button"
  >
    <button
      aria-label="button"
      className="button--send"
      type="button"
    >
      <a
        href="/filters"
        onClick={[Function]}
      >
        Encontrar
      </a>
    </button>
  </div>
  <div
    className="link"
  >
    <p>
      ¿Quieres Publicar?, 
      <a
        href="/register"
        onClick={[Function]}
      >
        Regístrate
      </a>
    </p>
  </div>
</section>
`;
